src/components/views/messages/RoomPredecessorTile.tsx

```diff
@@ -86,18 +88,32 @@ export const RoomPredecessorTile: React.FC<IProps> = ({ mxEvent, timestamp }) =>
     }
 
     const prevRoom = MatrixClientPeg.get().getRoom(predecessor.roomId);
-    if (!prevRoom) {
+
+    // We need either the previous room, or some servers to find it with.
+    // Otherwise, we must bail out here
+    if (!prevRoom && !predecessor.viaServers) {
```

Comment:
according to our style guide, we need two explicit casts to boolean here.

I think our style guide is mistaken on this point, so feel free to leave this and open a discussion on our style guide :)

Reply:
actually perhaps the `!` makes this acceptable? I don't really know what the style guide wants here.

Reply:
Definitely - `!!!foo` would be awful.